jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/StoreEngine.java

```diff
@@ -152,7 +152,7 @@ public synchronized boolean init(final StoreEngineOptions opts) {
             rOpts.setInitialServerList(opts.getInitialServerList());
             if (rOpts.getNodeOptions() == null) {
                 // copy common node options
-                rOpts.setNodeOptions(JRaftHelper.copyNodeOptionsFrom(opts.getCommonNodeOptions()));
+                rOpts.setNodeOptions(opts.getCommonNodeOptions().copy());
```

Comment:
null check

Reply:
I will resolve this issue and commit the resolved codes. @fengjiachun

Reply:
if commonNodeOptions is null,there maybe cause NPE.

Reply:
I look into the StoreEngineOptions class code,and find that commonNodeOptions is one of attributes for it.And commonNodeOptions is initilized by new object. @fengjiachun

Reply:
But when user call withCommonNodeOptions method of StoreEngineOptionsConfigured with null commonNodeOptions,this commonNodeOptions will be null.So on the safe side,null check maybe better.

Reply:
And origin copyNodeOptionsFrom logic also check commonNodeOptions as follows:
```
public static NodeOptions copyNodeOptionsFrom(final NodeOptions from) {
        final NodeOptions to = new NodeOptions();
        if (from == null) {
            return to;
        }
        to.setElectionTimeoutMs(from.getElectionTimeoutMs());
        to.setSnapshotIntervalSecs(from.getSnapshotIntervalSecs());
        to.setCatchupMargin(from.getCatchupMargin());
        to.setFilterBeforeCopyRemote(from.isFilterBeforeCopyRemote());
        to.setDisableCli(from.isDisableCli());
        to.setTimerPoolSize(from.getTimerPoolSize());
        to.setCliRpcThreadPoolSize(from.getCliRpcThreadPoolSize());
        to.setRaftRpcThreadPoolSize(from.getRaftRpcThreadPoolSize());
        to.setEnableMetrics(from.isEnableMetrics());
        to.setRaftOptions(copyRaftOptionsFrom(from.getRaftOptions()));
        return to;
    }
```